File: flutterapp/lib/utils/ImageUtils.dart
class ImageUtils {
  static String? getWeatherIconAsset(String weather) {
    final lowerWeather = weather.toLowerCase();

    if (lowerWeather.contains("rain")) {
      return "assets/weather_icons/ic_rain.png";
    } else if (lowerWeather.contains("partly")) {
      return "assets/weather_icons/ic_cloudy_day_1.png";
    } else if (lowerWeather.contains("mostly")) {
      return "assets/weather_icons/ic_cloudy_day.png";
    } else if (lowerWeather.contains("sunny")) {
      return "assets/weather_icons/ic_sun.png";
    } else if (lowerWeather.contains("snow")) {
      return "assets/weather_icons/ic_snowing_1.png";
    }

    return null;
  }
}

File: archive/androidapp/app/src/main/java/com/example/lanweather/ui/utils/ImageUtils.kt
package com.example.lanweather.ui.utils

import android.content.Context
import android.content.res.Resources
import android.widget.ImageView
import androidx.core.content.res.ResourcesCompat
import com.example.lanweather.R

class ImageUtils {
    companion object {
        fun setWeatherIcon(weather: String, imageView: ImageView, resources: Resources, context: Context?) {
            when {
                weather.toLowerCase().contains("rain") -> {
                    imageView.setImageDrawable(ResourcesCompat.getDrawable(resources, R.drawable.ic_rain, context?.theme))
                }
                weather.toLowerCase().contains("partly") -> {
                    imageView.setImageDrawable(ResourcesCompat.getDrawable(resources, R.drawable.ic_cloudy_day_1, context?.theme))
                }
                weather.toLowerCase().contains("mostly") -> {
                    imageView.setImageDrawable(ResourcesCompat.getDrawable(resources, R.drawable.ic_cloudy_day, context?.theme))
                }
                weather.toLowerCase().contains("sunny") -> {
                    imageView.setImageDrawable(ResourcesCompat.getDrawable(resources, R.drawable.ic_sun, context?.theme))
                }
                weather.toLowerCase().contains("snow") -> {
                    imageView.setImageDrawable(ResourcesCompat.getDrawable(resources, R.drawable.ic_snowing_1, context?.theme))
                }
            }
        }
    }
}